assets/src/edit-story/components/canvas/canvasUploadDropTarget.js

```diff
@@ -48,18 +49,91 @@ const PageAreaCover = styled(PageArea)`
 `;
 
 function CanvasUploadDropTarget({ children }) {
-  const { uploadFile } = useUploader();
+  const { uploadFile, sizeErrorMessage, validErrorMessage } = useUploader();
+  const { createSnackbar } = useSnackbar();
   const insertElement = useInsertElement();
   const onDropHandler = useCallback(
-    (files) => {
-      files.forEach((file) => {
-        uploadFile(file).then((res) => {
+    async (files) => {
+      const possibleRetryFiles = [];
+      const sizeErrorFiles = [];
+      const validErrorFiles = [];
+      const otherErrorFiles = [];
+      const isMultiple = files.length > 1;
+      for (const file of files) {
+        try {
+          // eslint-disable-next-line no-await-in-loop
+          const res = await uploadFile(file);
           const resource = getResourceFromUploadAPI(res);
           insertElement(resource.type, { resource });
+        } catch (e) {
+          if (e.name !== 'SizeError' && e.name !== 'ValidError') {
```

Comment:
This snippet of code is repeated https://github.com/google/web-stories-wp/pull/713/files#diff-c9756545ef6a994f3ba0091cf289a47fR169 . Can we just create a function for this so it can be reused?